code/datums/controllers/action_controls.dm

```diff
@@ -673,6 +673,10 @@ var/datum/action_controller/actions
 			boutput(owner, "<span class='notice'>You have nothing to build with!</span>")
 			interrupt(INTERRUPT_ALWAYS)
 			return
+		if(locate(/obj/machinery/atmospherics) in owner.loc)
+			boutput(owner, "<span class='notice'>You cant build another pipe there!</span>")
+			interrupt(INTERRUPT_ALWAYS)
+			return
```

Comment:
Wouldn't this check happen regardless of what you're crafting?

Reply:
fixed